docs/solidity-to-cadence.mdx

```diff
@@ -0,0 +1,447 @@
+
+## Account key formulation
+
+In EVM-based chains, an address is derived from a cryptographically generated public key and can have a single private
+key, supporting one type of signature curve, i.e. ECDSA. They are not verifiable off-chain and typos/truncation in an
+address may result in funds being lost.
+
+Flow account addresses have a special format and are verifiable off-chain. Verifying address format validity can be
+done using an error detection algorithm based on linear code. While this does not also confirm that an address is active
+on-chain the extra verifiability is a useful safeguard.
+
+## Contract size constraints
+
+Solidity developers will be well aware of the [EIP-170](https://eips.ethereum.org/EIPS/eip-170) deployable contract
+bytecode size limit of 24KB. This can burden builders who need to optimize contract bytecode size, sometimes even
+requiring a re-design of contracts to break it into smaller contract parts.
+
+By contrast, Cadence has no inherent or defined smart contract size limit. However, it is restricted by the transaction
+size limit which is 1.5MB. With very rare exceptions, it’s unlikely that this limit would pose a problem to those
+developing Cadence contracts. Should it be needed, there is a known way to deploy a contract exceeding 1.5 MB which we
+will document at a later time.
+
+# Low level language differences
+
+## Arithmetic
+
+Historically, Solidity, smart contracts lost millions of dollars because of improper handling of arithmetic
+under/overflows. Contemporary Solidity versions offer inbuilt handling of under/overflow for arithmetic operations.
+
+Cadence implements [saturating math](https://en.wikipedia.org/wiki/Saturation_arithmetic) that avoids
+overflow/underflow.
+
+## Optional support
+
+[Optional bindings](language#control-flow) provide in-built conditional handling of nil values. Optionals have two
+cases: either there is a value, or there is nothing; they fork program flow based on whether a value is nil or not.
+
+## Iterable Dictionaries
+
+Solidity offers the mapping type, however, it is not iterable. Because of that dApp developers have to run off-chain
+services to know all the keys to the mapping. This also pushes builders to create custom datatypes like EnumerableMap
+which adds to gas costs.
+
+Cadence offers the [Dictionary](language#control-flow) type, an unordered collection of key-value associations
+which is iterable.
+
+## Rich support for type utility functions
+
+Cadence offers numerous native-type utility functions to simplify development. For example, the String type
+provides:
+- utf8
+- concat()
+- slice()
+- decodeHex()
+- encodeHex()
+- toLower()
+- length
+
+## Argument labelling
+
+Argument labels in Cadence help to disambiguate input values. They makes code more readable and explicit. They
+also eliminate confusion around the order of arguments when working with the same type.
+
+Eg:
+`fun foo(balance: UFix64)`
+called as
+`self.foo(balance: 30.0)`
```

Comment:
```suggestion
`self.foo(balance: 30.0)`

`fun foo( _balance: UFix64)`
can be called as
`self.foo(balance: 30.0)`
or as
`self.foo(30.0)`
```

Maybe this is going too much into detail on cadence syntax, however I'll make an extra emphasis on that argument labels are mandatory when calling the function by default